element_facemap/facemap_inference.py

```diff
@@ -0,0 +1,496 @@
+            subject/sessions. Returns a string for the full path to the root data directory.
+        + get_facemap_processed_data_dir(session_key: dict) -> str
+            Optional function to retrieve the desired output directory
+            for Facemap files for a given session. If unspecified,
+            the output is stored in the video folder for the session, which is the default behavior of Facemap.
+            Returns a string of the absolute path of the output directory.
+    """
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'dependency' must be a module's name or a module"
+    assert hasattr(
+        linking_module, "get_facemap_root_data_dir"
+    ), "The linking module must specify a lookup function for a root data directory"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    # activate facial behavioral extimation (fbe) schema
+    fbe.activate(
+        fbe_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+
+    # activate facial pose model schema
+    schema.activate(
+        facemap_model_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+# ----------------------------- Table declarations ----------------------
+@schema
+class BodyPart(dj.Lookup):
+    """Body parts tracked by Facemap models
+
+    Attributes:
+        body_part ( varchar(32) ): Body part short name.
+        body_part_description ( varchar(1000),optional ): Full description
+
+    """
+
+    definition = """
+    body_part                : varchar(32)
+    ---
+    body_part_description='' : varchar(1000)
+    """
+
+    # Facemap Default BodyPart list
+    contents = [
+        ("eye(back)", ""),
+        ("eye(bottom)", ""),
+        ("eye(front)", ""),
+        ("eye(top)", ""),
+        ("lowerlip", ""),
+        ("mouth", ""),
+        ("nose(bottom)", ""),
+        ("nose(r)", ""),
+        ("nose(tip)", ""),
+        ("nose(top)", ""),
+        ("nosebridge", ""),
+        ("paw", ""),
+        ("whisker(I)", ""),
+        ("whisker(III)", ""),
+        ("whisker(II)", ""),
+    ]
+
+
+@schema
+class FacemapModel(dj.Manual):
+    """Trained Models stored for facial pose inference
+
+    Attributes:
+        model_id(int) : User specified ID associated with a unique model
+        model_name( varchar(64) ): Name of model, filepath.stem
+    """
+
+    definition = """
+    model_id                  : int              # user assigned ID associated with a unique model
+    ---
+    model_name                : varchar(64)      # name of model 
+    model_description=''      : varchar(1000)    # optional model description
+    """
+
+    class BodyPart(dj.Part):
```

Comment:
This table could also be renamed `KeyPoints` if the Lookup table above is renamed